compliance_checker/cfunits.py

```diff
@@ -0,0 +1,90 @@
+from pyudunits2 import UnitSystem, UnresolvableUnitException
+
+try:
+    import cf_units
+except ImportError:
+    cf_units = False
+
+
+class PyUdunits2:
+    """Workaround for the differences in pyudunits2 and cf-units.
+
+    NB: Some of these may change and/or get implemented upstream. Pyudunits2 is new and in-flux.
+
+    1/4 Raise the same ValueError to match cf-unit errors.
+    2/4 Creates an empty unit from None to mimic cf-unit's Unit('unknown')
+    3/4 Add a definition object that is ust units.expanded()
+    """
+
+    def __init__(self, units):
+        """Keep unit system so we can convert from string later."""
+        self.ut_system = UnitSystem.from_udunits2_xml()
```

Comment:
This will be the source of slowdown. We definitely don't want to do this for every unit we are creating - it isn't cached anywhere at this point. Perhaps simply:

```
def __init__(...):
    global UT_SYSTEM
    if UT_SYSTEM is None:
        UT_SYSTEM = UnitSystem.from_udunits2_xml()
```